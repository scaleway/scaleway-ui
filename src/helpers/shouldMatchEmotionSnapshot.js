/* eslint-disable import/no-extraneous-dependencies */
import createCache from '@emotion/cache'
import { createSerializer } from '@emotion/jest'
import { CacheProvider, ThemeProvider } from '@emotion/react'
import { render } from '@testing-library/react'
import React from 'react'
import theme from '../theme'

expect.addSnapshotSerializer(createSerializer())

const emotionCache = createCache({
  key: 'cache',
})
emotionCache.compat = true

export default async (component, { transform } = {}) => {
  const node = render(
    <CacheProvider value={emotionCache}>
      <ThemeProvider theme={theme}>{component}</ThemeProvider>
    </CacheProvider>,
  )
  if (transform) await transform(node)

  expect(node.asFragment()).toMatchSnapshot()
}
