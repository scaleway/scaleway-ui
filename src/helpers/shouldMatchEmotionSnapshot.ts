/* eslint-disable import/no-extraneous-dependencies */
import { createSerializer } from '@emotion/jest'
import { RenderOptions, render } from '@testing-library/react'
import { ReactNode } from 'react'
import defaultTheme from '../theme'
import renderWithTheme from './renderWithTheme'

// use only class hash (generated from css style content)
expect.addSnapshotSerializer(
  createSerializer({ classNameReplacer: className => className }),
)

interface Options {
  options?: RenderOptions
  theme?: typeof defaultTheme
  transform?: (node: ReturnType<typeof render>) => Promise<void> | void
}

export default async (
  component: ReactNode,
  { transform, options, theme }: Options = {},
): Promise<void> => {
  const node = renderWithTheme(component, options, theme)
  if (transform) await transform(node)

  expect(node.asFragment()).toMatchSnapshot()
  node.unmount()
}
