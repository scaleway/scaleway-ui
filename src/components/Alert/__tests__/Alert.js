import React from 'react'
import { Alert } from '..'
import shouldMatchEmotionSnapshot from '../../../helpers/shouldMatchEmotionSnapshot'

test('Alert renders correctly ', () => {
  shouldMatchEmotionSnapshot(<Alert />)
})

test('Alert renders with variant success', () => {
  shouldMatchEmotionSnapshot(<Alert vairant="success" />)
})

test('Alert renders with variant warning', () => {
  shouldMatchEmotionSnapshot(<Alert vairant="warning" />)
})

test('Alert renders with variant info', () => {
  shouldMatchEmotionSnapshot(<Alert vairant="info" />)
})

test('Alert renders with variant success & icon lock', () => {
  shouldMatchEmotionSnapshot(<Alert vairant="info" icon="lock" />)
})
