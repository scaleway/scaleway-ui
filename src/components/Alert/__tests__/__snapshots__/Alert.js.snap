// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert Alert renders correctly  1`] = `
.emotion-1 {
  color: #4a4f62;
  font-weight: 400;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 16px;
  color: #ef5774;
  font-weight: 700;
}

.emotion-0 {
  fill: #ef5774;
  vertical-align: middle;
  height: 40px;
  width: 40px;
  min-width: 40px;
  min-height: 40px;
}

.emotion-0.emotion-0 {
  margin-right: 16px;
}

<div
  className="sui-box emotion-1"
  role="alert"
>
  <svg
    className="sui-box sc-ui-icon emotion-0"
    viewBox="0 0 24 24"
  >
    <path
      d="M13,14H11V10H13M13,18H11V16H13M1,21H23L12,2L1,21Z"
    />
  </svg>
  <span />
</div>
`;

exports[`Alert Alert renders with variant info 1`] = `
.emotion-1 {
  color: #4a4f62;
  font-weight: 400;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 16px;
  color: #ef5774;
  font-weight: 700;
}

.emotion-0 {
  fill: #ef5774;
  vertical-align: middle;
  height: 40px;
  width: 40px;
  min-width: 40px;
  min-height: 40px;
}

.emotion-0.emotion-0 {
  margin-right: 16px;
}

<div
  className="sui-box emotion-1"
  role="alert"
  vairant="info"
>
  <svg
    className="sui-box sc-ui-icon emotion-0"
    viewBox="0 0 24 24"
  >
    <path
      d="M13,14H11V10H13M13,18H11V16H13M1,21H23L12,2L1,21Z"
    />
  </svg>
  <span />
</div>
`;

exports[`Alert Alert renders with variant success & icon lock 1`] = `
.emotion-1 {
  color: #4a4f62;
  font-weight: 400;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 16px;
  color: #ef5774;
  font-weight: 700;
}

.emotion-0 {
  fill: #ef5774;
  vertical-align: middle;
  height: 40px;
  width: 40px;
  min-width: 40px;
  min-height: 40px;
}

.emotion-0.emotion-0 {
  margin-right: 16px;
}

<div
  className="sui-box emotion-1"
  role="alert"
  vairant="info"
>
  <svg
    className="sui-box sc-ui-icon emotion-0"
    viewBox="0 0 24 24"
  >
    <path
      d="M12,17A2,2 0 0,0 14,15C14,13.89 13.1,13 12,13A2,2 0 0,0 10,15A2,2 0 0,0 12,17M18,8A2,2 0 0,1 20,10V20A2,2 0 0,1 18,22H6A2,2 0 0,1 4,20V10C4,8.89 4.9,8 6,8H7V6A5,5 0 0,1 12,1A5,5 0 0,1 17,6V8H18M12,3A3,3 0 0,0 9,6V8H15V6A3,3 0 0,0 12,3Z"
    />
  </svg>
  <span />
</div>
`;

exports[`Alert Alert renders with variant success 1`] = `
.emotion-1 {
  color: #4a4f62;
  font-weight: 400;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 16px;
  color: #ef5774;
  font-weight: 700;
}

.emotion-0 {
  fill: #ef5774;
  vertical-align: middle;
  height: 40px;
  width: 40px;
  min-width: 40px;
  min-height: 40px;
}

.emotion-0.emotion-0 {
  margin-right: 16px;
}

<div
  className="sui-box emotion-1"
  role="alert"
  vairant="success"
>
  <svg
    className="sui-box sc-ui-icon emotion-0"
    viewBox="0 0 24 24"
  >
    <path
      d="M13,14H11V10H13M13,18H11V16H13M1,21H23L12,2L1,21Z"
    />
  </svg>
  <span />
</div>
`;

exports[`Alert Alert renders with variant warning 1`] = `
.emotion-1 {
  color: #4a4f62;
  font-weight: 400;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 16px;
  color: #ef5774;
  font-weight: 700;
}

.emotion-0 {
  fill: #ef5774;
  vertical-align: middle;
  height: 40px;
  width: 40px;
  min-width: 40px;
  min-height: 40px;
}

.emotion-0.emotion-0 {
  margin-right: 16px;
}

<div
  className="sui-box emotion-1"
  role="alert"
  vairant="warning"
>
  <svg
    className="sui-box sc-ui-icon emotion-0"
    viewBox="0 0 24 24"
  >
    <path
      d="M13,14H11V10H13M13,18H11V16H13M1,21H23L12,2L1,21Z"
    />
  </svg>
  <span />
</div>
`;
